### ISSUE TITLE
web.config runtime section Merge issue

### ISSUE THREAD

**Issue by paratoner** [loftum/ConfigMerge#1]
Hi,
Thank you very much for this project.
I have two web.config file that contains <runtime> section like this 
web1.config file : 
```
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="System.Web.Helpers" publicKeyToken="31bf3856ad364e35" />
        <bindingRedirect oldVersion="1.0.0.0-2.0.0.0" newVersion="2.0.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Web.Mvc" publicKeyToken="31bf3856ad364e35" />
        <bindingRedirect oldVersion="1.0.0.0-5.1.0.0" newVersion="5.0.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Web.WebPages" publicKeyToken="31bf3856ad364e35" />
        <bindingRedirect oldVersion="1.0.0.0-2.0.0.0" newVersion="2.0.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Autofac" publicKeyToken="17863af14b0044da" culture="neutral" />
        Assembly versions can be redirected in app,
        publisher policy, or machine configuration files.
        <bindingRedirect oldVersion="0.0.0.0-3.5.0.0" newVersion="3.5.0.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
```
web2.config file 
```
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="Autofac" publicKeyToken="17863af14b0044da"  culture="neutral" />
        Assembly versions can be redirected in app,
        publisher policy, or machine configuration files.
        <bindingRedirect oldVersion="0.0.0.0-3.4.0.0" newVersion="3.5.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Autofac.Integration.Web" publicKeyToken="17863af14b0044da"  culture="neutral" />
        Assembly versions can be redirected in app,
        publisher policy, or machine configuration files.
        <bindingRedirect oldVersion="0.0.0.0-3.0.0.0" newVersion="3.2.0.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
```

and when i call ConfigMerge.exe from commandline
`ConfigMerge.exe -recipe:merged.config=[Web1.config,Web2.config];`

Error occurs like this 

```
ConfigMerge: error: Cannot merge into file containing multiple undelimited equal
 siblings, needs attribute with name: id, name, key, path, virtualPath, href, na
mespace or xmlns. The offender is: <assemblyBinding xmlns="urn:schemas-microsoft
-com:asm.v1">
  <dependentAssembly>
    <assemblyIdentity name="Autofac" publicKeyToken="17863af14b0044da" culture="
neutral" />
        Assembly versions can be redirected in app,
        publisher policy, or machine configuration files.
        <bindingRedirect oldVersion="0.0.0.0-3.4.0.0" newVersion="3.5.0.0" /></d
ependentAssembly>
  <dependentAssembly>
    <assemblyIdentity name="Autofac.Integration.Web" publicKeyToken="17863af14b0
044da" culture="neutral" />
        Assembly versions can be redirected in app,
        publisher policy, or machine configuration files.
        <bindingRedirect oldVersion="0.0.0.0-3.0.0.0" newVersion="3.2.0.0" /></d
ependentAssembly>
</assemblyBinding>
   at ConfigMerge.Core.Merging.XmlMerge.Merge(XElement template, XElement input)
 in D:\Bana Özel\Experimental\ConfigMerge-master\ConfigMerge-master\ConfigMerge\
Core\Merging\XmlMerge.cs:line 77
   at ConfigMerge.Core.Merging.XmlMerge.Merge(XElement template, XElement input)
 in D:\Bana Özel\Experimental\ConfigMerge-master\ConfigMerge-master\ConfigMerge\
Core\Merging\XmlMerge.cs:line 113
   at ConfigMerge.Core.Merging.XmlMerge.Merge(XElement template, XElement input)
 in D:\Bana Özel\Experimental\ConfigMerge-master\ConfigMerge-master\ConfigMerge\
Core\Merging\XmlMerge.cs:line 82
   at ConfigMerge.Core.Merging.XmlMerge.Merge(XElement input) in D:\Bana Özel\Ex
perimental\ConfigMerge-master\ConfigMerge-master\ConfigMerge\Core\Merging\XmlMer
ge.cs:line 27
   at ConfigMerge.Core.Merging.XmlMerge..ctor(XDocument result, XmlSource source
, TransformOptions options) in D:\Bana Özel\Experimental\ConfigMerge-master\Conf
igMerge-master\ConfigMerge\Core\Merging\XmlMerge.cs:line 22
   at ConfigMerge.Core.Merging.XmlMerger.Merge(IEnumerable`1 inputs) in D:\Bana
Özel\Experimental\ConfigMerge-master\ConfigMerge-master\ConfigMerge\Core\Merging
\XmlMerger.cs:line 20
   at ConfigMerge.Core.ConfigTransformer.DoTransform(String outputPath, IEnumera
ble`1 inputPaths) in D:\Bana Özel\Experimental\ConfigMerge-master\ConfigMerge-ma
ster\ConfigMerge\Core\ConfigTransformer.cs:line 53
   at ConfigMerge.Core.ConfigTransformer.Transform(String outputFile, IEnumerabl
e`1 inputFiles) in D:\Bana Özel\Experimental\ConfigMerge-master\ConfigMerge-mast
er\ConfigMerge\Core\ConfigTransformer.cs:line 32
   at lambda_method(Closure , IConfigTransformer )
   at ConfigMerge.Program.Main(String[] args) in D:\Bana Özel\Experimental\Confi
gMerge-master\ConfigMerge-master\ConfigMerge\Program.cs:line 35
```

**Comment by loftum** [comment]
Glad you like the project!
And thank you for the well-documented issue.

TL;DR The issue is not a bug, but a lack of a feature which would have been nice to have.

The xml merger is a simple one, and was designed mainly for replacing appSettings. For any replace-operation, the xml  (child) node in question will need some identifying attribute, like id, name, etc.
In your case, what the error message is really saying, is that <dependentAssembly> is lacking an identifying attribute, and that it therefore does not know how to merge the 2 <assemblyBinding> sections in question.

An <assemblyBinding> is a complex config element identified by a child <assemblyIdentity>. To merge assemblyBindings we would need to either rethink how to identify elements in general, or write custom code for <assemblyBinding>. And I would really like to avoid the latter, if possible.

I am currently on vacation, and am not able to look into it at the moment. But if you want to hack a solution yourself, this is probably where you would want to start:
https://github.com/loftum/ConfigMerge/blob/master/ConfigMerge/Core/Merging/XmlMerge.cs#L75

Argument template would represent <assemblyBinding> from web1.config and input would represent <assemblyBinding> from web2.config.

**Comment by loftum** [comment]
I have added support for nested identifiers in TransformOptions:

`UniqueAttributes = new[] { "id", "name", "key", "path", "virtualPath", "href", "namespace", "dependentAssembly/assemblyIdentity/name" };`
This means that any `dependentAssembly` node will be identified by attribute `name `from its `assemblyIdentity `child node, and are now mergable.
Note that UniqueAttributes can be overridden in App.config (ConfigMerge.exe.config)

This feature is currently available in branch branch issue-1-runtime-section, and I added `AssemblyBindingOverrideTest` which just console outs a simple test.

Feel free to check it out!

**Comment by loftum** [comment]
I have added tests (AssemblyBindingMergeTest) and merged the changes to master.
